File: js/task2b/S2_FOR_03.js
function sumUntil(nums, threshold) {
  let sum = 0;

  for (let num of nums) {
      if (sum + num > threshold) {
          break;
      }
      sum += num;
  }

  return sum;
}

console.log(sumUntil([1, 2, 3, 4, 5], 7));    
console.log(sumUntil([1, 2, 3, 4, 5], 10));   
console.log(sumUntil([1, 2, 3, 4, 5], 3));   
console.log(sumUntil([1, 2, 3, 4, 5], 0));   
console.log(sumUntil([], 5));                 

File: python/task2b/S2_FOR_03.py
def sumUntil(nums, threshold):
  total = 0
  for num in nums:
      if total + num > threshold:
          break
      total += num
  return total

print(sumUntil([1, 2, 3, 4, 5], 7))
print(sumUntil([1, 2, 3, 4, 5], 10))
print(sumUntil([1, 2, 3, 4, 5], 3))
print(sumUntil([1, 2, 3, 4, 5], 0))
print(sumUntil([], 5))
